streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -200,20 +200,23 @@ private void closeAndRevive(final Map<Task, Collection<TopicPartition>> taskWith
             }
             task.closeDirty();
 
+            removedTasks.put((AbstractTask) task, task.changelogPartitions());
+            ((AbstractTask) task).stateMgr.clear();
```

Comment:
This seems not good...we shouldn't be casting to an abstract class. Let's just add whatever we need as a method on the `Task` interface